app/controllers/agenda-items/index.ts

```diff
@@ -46,34 +145,55 @@ export default class AgendaItemsIndexController extends Controller {
     }
   };
 
-  get params(): AgendaItemsParams {
+  get filters(): AgendaItemsParams {
     return {
       keyword: this.keyword,
       municipalityLabels: this.municipalityLabels,
       plannedStartMin: this.plannedStartMin,
       plannedStartMax: this.plannedStartMax,
     };
   }
-
-  @action
-  async loadMore() {
-    if (!this.isLoadingMore) {
-      this.isLoadingMore = true;
-
-      this.currentPage++;
-
-      const agendaItems = await getAgendaItems(
-        this,
-        this.params,
-        this.currentPage
-      );
-      this.agendaItems = [...this.agendaItems, ...agendaItems];
-
-      this.isLoadingMore = false;
-    }
-  }
-
-  get municipalities() {
-    return this.municipalityList.municipalities();
-  }
 }
+
+const agendaItemsQuery = ({
+  page,
+  keyword,
+  locationIds,
+  plannedStartMin,
+  plannedStartMax,
+}: {
+  page: number;
+  keyword?: string;
+  locationIds?: string;
+  plannedStartMin?: string;
+  plannedStartMax?: string;
+}) => ({
+  include: [
+    'sessions.governing-body.is-time-specialization-of.administrative-unit.location',
+    'sessions.governing-body.administrative-unit.location',
+  ].join(','),
+  sort: '-sessions.planned-start',
+  filter: {
+    sessions: {
+      ':gt:planned-start': plannedStartMin ? plannedStartMin : undefined,
```

Comment:
It would probably be better to simply not add the filters if the value isn't set instead of using `undefined`. I'm not sure if this would mean less work for mu-cl-resources, but if it does it potentially increases the speed. Since mu-auth is coming it might not be worth tinkering with it though.

Reply:
When value is `undefined`, it is not part of the request. I guess it's filtered by `ember-data`.

Reply:
TIL 👍 I could sworn I've seen it output `undefined` somewhere in requests before 😄.